sbncode/FlashMatch/FlashPredict_module.cc

```diff
@@ -917,18 +933,60 @@ FlashPredict::FlashMetrics FlashPredict::computeFlashMetrics(
   }
   else {
     std::string channels;
-    for(auto oph=opH_beg; oph!=opH_end; ++oph) channels += std::to_string(oph->OpChannel()) + ' ';
+    for(auto oph : ophits) channels += std::to_string(oph.OpChannel()) + ' ';
     mf::LogError("FlashPredict")
       << "Really odd that I landed here, this shouldn't had happen.\n"
       << "sum:          \t" << sum << "\n"
       << "sum_PE:       \t" << sum_PE << "\n"
       << "sum_unPE:     \t" << sum_unPE << "\n"
-      << "opHits size:  \t" << std::distance(opH_beg, opH_end) << "\n"
+      << "opHits size:  \t" << ophits.size() << "\n"
       << "channels:     \t" << channels << std::endl;
     return {};
   }
 }
 
+FlashPredict::FlashMetrics FlashPredict::getFlashMetrics(
+  const FlashPredict::SimpleFlash& sf) const
+{
+  std::vector<recob::OpHit> ophits;
+  const OpHitIt oph_beg = sf.opH_beg;
+  const OpHitIt oph_end = sf.opH_end;
+  std::copy(oph_beg, oph_end, std::back_inserter(ophits));
+
+  auto fm = computeFlashMetrics(ophits);
+  fm.id = sf.flashId;
+  fm.time = sf.maxpeak_time;
+  fm.activity = sf.ophsInVolume;
+  return fm;
+}
+
+FlashPredict::FlashMetrics FlashPredict::getFlashMetrics(
+  const recob::OpFlash& opflash,
+  std::vector<art::Ptr<recob::OpHit>> ophit_v,
+  unsigned id) const
+{
+  std::vector<recob::OpHit> ophits;
+  for(unsigned i=0; i< ophit_v.size(); i++) {
+    ophits.emplace_back(*(ophit_v)[i]);
+  }
+
+  auto fm = computeFlashMetrics(ophits);
+  fm.id = id;
+  if(fSBND) fm.time = opflash.AbsTime();
+  else if(fICARUS) fm.time = opflash.Time();
+  fm.activity = (opflash.XCenter() < 0) ? 100 : 200;
+  if(fUseOpCoords) {
```

Comment:
I feel that if you really want to use the OpFlashCoords then you should overwrite the corresponding `fm`variables (`fm.x, fm.y, fm.z, ...`).

Currently the `fm.op...` vars are not used anywhere else, they are written here but never read or used anywhere else.